packages/ra-ui-materialui/src/list/ListNoResults.tsx

```diff
@@ -2,15 +2,20 @@ import * as React from 'react';
 import { memo } from 'react';
 import CardContent from '@mui/material/CardContent';
 import Typography from '@mui/material/Typography';
-import { useResourceContext, useTranslate } from 'ra-core';
+import { useListController, useResourceContext, useTranslate } from 'ra-core';
+import { Link } from '@mui/material';
 
 export const ListNoResults = memo(() => {
     const translate = useTranslate();
     const resource = useResourceContext();
+    const { setFilters } = useListController({ resource });
```

Comment:
`useListController`, not `useListContext`! 

I fixed the mistake in 199fbaec041f6c5ded6f78bbc0eb4d3d2b4c8069